src/sealed.rs

```diff
@@ -32,3 +34,9 @@ impl Sealed for Bound<'_, PyString> {}
 impl Sealed for Bound<'_, PyTraceback> {}
 impl Sealed for Bound<'_, PyTuple> {}
 impl Sealed for Bound<'_, PyType> {}
+
+impl<T> Sealed for AddTypeToModule<T> {}
+impl<T> Sealed for AddClassToModule<T> {}
```

Comment:
I had a quick reflection on whether the generic implementations here are a problem; I don't think they are as we control these types and they're hidden zero-sized types with no runtime use.

Note that something like `impl<T> Sealed for Vec<T> {}` would be a problem as users could then start adding other PyO3 trait implementations to `Vec<MyType>`, which they could then control and do stuff with.